packages/chat-stateful-client/src/StatefulChatThreadClient.ts

```diff
@@ -102,21 +108,27 @@ class ProxyChatThreadClient implements ProxyHandler<ChatThreadClient> {
         return createDecoratedListReadReceipts(chatThreadClient, this._context);
       }
       case 'removeParticipant': {
-        return async (...args: Parameters<ChatThreadClient['removeParticipant']>) => {
-          const result = await chatThreadClient.removeParticipant(...args);
-          const [removeIdentifier] = args;
-          this._context.deleteParticipant(chatThreadClient.threadId, getIdentifierKind(removeIdentifier));
-          return result;
-        };
+        return this._context.withAsycnErrorTeedToState(
+          async (...args: Parameters<ChatThreadClient['removeParticipant']>) => {
+            const result = await chatThreadClient.removeParticipant(...args);
+            const [removeIdentifier] = args;
+            this._context.deleteParticipant(chatThreadClient.threadId, getIdentifierKind(removeIdentifier));
+            return result;
+          },
+          'ChatThreadClient.removeParticipant'
+        );
       }
       case 'updateMessage': {
-        return async (...args: Parameters<ChatThreadClient['updateMessage']>) => {
-          const result = await chatThreadClient.updateMessage(...args);
-          const [messageId, updateOption] = args;
+        return this._context.withAsycnErrorTeedToState(
+          async (...args: Parameters<ChatThreadClient['updateMessage']>) => {
+            const result = await chatThreadClient.updateMessage(...args);
+            const [messageId, updateOption] = args;
 
-          this._context.updateChatMessageContent(chatThreadClient.threadId, messageId, updateOption?.content);
-          return result;
-        };
+            this._context.updateChatMessageContent(chatThreadClient.threadId, messageId, updateOption?.content);
+            return result;
+          },
+          'ChatThreadClient.updateMessage'
+        );
       }
       case 'updateTopic': {
         return this._context.withAsycnErrorTeedToState(async (...args: Parameters<ChatThreadClient['updateTopic']>) => {
```

Comment:
withAsyncErrorTeedToState

Reply:
Done. This was a typo lower in the stack.